googlebigtable2/src/main/java/site/ycsb/db/GoogleBigtable2Client.java

```diff
@@ -0,0 +1,420 @@
+  private static final String ENDPOINT_KEY = PROP_PREFIX + ".data-endpoint";
+  private static final String PROJECT_KEY = PROP_PREFIX + ".project";
+  private static final String INSTANCE_KEY = PROP_PREFIX + ".instance";
+  private static final String APP_PROFILE_ID_KEY = PROP_PREFIX + ".app-profile";
+  private static final String FAMILY_KEY = PROP_PREFIX + ".family";
+
+  private static final String MAX_OUTSTANDING_BYTES = PROP_PREFIX + ".max-outstanding-bytes";
+  private static final String CLIENT_SIDE_BUFFERING = PROP_PREFIX + ".use-batching";
+
+  /**
+   * Print debug information to standard out.
+   */
+  private static boolean debug = false;
+
+  /**
+   * Tracks running thread counts so we know when to close the session.
+   */
+  private static int clientRefCount = 0;
+
+  /**
+   * Global Bigtable native API objects.
+   */
+  private static BigtableDataClient client;
+
+  private static String columnFamily;
+  /**
+   * If true, buffer mutations on the client. For measuring insert/update/delete latencies, client
+   * side buffering should be disabled.
+   */
+  private static boolean clientSideBuffering = false;
+
+  /**
+   * Thread local Bigtable native API objects.
+   */
+  private final Map<String, Batcher<RowMutationEntry, Void>> batchers = new HashMap<>();
+
+  @Override
+  public void init() throws DBException {
+    Properties props = getProperties();
+
+    try {
+      globalInit(props);
+    } catch (Exception e) {
+      throw new DBException("Failed to initialize client", e);
+    }
+  }
+
+  private static synchronized void globalInit(Properties props) throws IOException {
+    clientRefCount++;
+    if (clientRefCount > 1) {
+      return;
+    }
+
+    debug = Boolean.parseBoolean(props.getProperty(DEBUG_KEY, "false"));
+
+    // Resource names
+    BigtableDataSettings.Builder builder =
+        BigtableDataSettings.newBuilder()
+            .setProjectId(getRequiredProp(props, PROJECT_KEY))
+            .setInstanceId(getRequiredProp(props, INSTANCE_KEY));
+
+    Optional.ofNullable(props.getProperty(APP_PROFILE_ID_KEY)).ifPresent(builder::setAppProfileId);
+
+    columnFamily = getRequiredProp(props, FAMILY_KEY);
+
+    // Endpoint
+    Optional<String> emulatorHost =
+        Optional.ofNullable(System.getenv().get("BIGTABLE_EMULATOR_HOST"));
+    Optional<String> dataEndpoint = Optional.ofNullable(props.getProperty(ENDPOINT_KEY));
+    if (emulatorHost.isPresent()) {
+      Preconditions.checkState(
+          !dataEndpoint.equals(emulatorHost),
+          "Can't override endpoint when BIGTABLE_EMULATOR_HOST is set");
+    } else {
+      dataEndpoint.ifPresent(ep -> builder.stubSettings().setEndpoint(ep));
+    }
+
+    // Other settings
+    clientSideBuffering =
+        Optional.ofNullable(props.getProperty(CLIENT_SIDE_BUFFERING))
+            .map(Boolean::parseBoolean)
+            .orElse(false);
+
+    Optional.ofNullable(props.getProperty(MAX_OUTSTANDING_BYTES))
```

Comment:
why making max outstadning bytes configurable?

Reply:
I think patrick overrides this in internal tests